cobalt/browser/metrics/cobalt_metrics_logs_uploader.h

```diff
@@ -47,12 +47,14 @@ class CobaltMetricsLogUploader : public metrics::MetricsLogUploader {
     return weak_factory_.GetWeakPtr();
   }
 
-  void setOnUploadComplete(
+  // Virtual for overriding in tests.
+  virtual void setOnUploadComplete(
       const ::metrics::MetricsLogUploader::UploadCallback& on_upload_complete) {
     on_upload_complete_ = on_upload_complete;
   }
 
-  void SetMetricsListener(
+  // Virtual for overriding in tests.
```

Comment:
nit: As a general rule, is best to try to not make methods
virtual just for testing. Instead, Chromium likes to add
setters with a `...ForTesting()` suffix (there's some python
magic that removes them from the build, and makes sure they
are only called from test code; the idiom is usually to
inject some fake or mock that otherwise would be internal.
Not sure if it applies here, though.

Reply:
Let's chat about this. I'm really struggling to see how I can replicate what's currently being tested in CobaltMetricsServiceClientTest using "ForTesting" methods. The virtuals allowing both assertion a function is called in the real implementation and also that its arguments are correct.  Not seeing a way to replicate that with "ForTesting" methods.